src/index.ts

```diff
@@ -168,7 +168,9 @@ export default class LocomotiveScroll {
         // Destroy Lenis
         this.lenisInstance.destroy();
         // Destroy Core
-        this.coreInstance.destroy();
+        requestAnimationFrame(() => {
+            this.coreInstance?.destroy();
+        });
```

Comment:
Since `useEffect` will run twice when we don't provide any values, it means that it will indeed hit the `destroy` at init. However, since we're waiting for 1 frame to initialize the core instance, it will be undefined in this case. Your suggestion is good, but I would also keep the destroy outside of the RAF.

```suggestion
      // Destroy Core
      this.coreInstance?.destroy();
      
      // Ensure a delay before destroying to handle cases of instant destruction
      requestAnimationFrame(() => {
          this.coreInstance?.destroy();
      });
```

Reply:
Is there any reason why destroying it immediately is critical? I don't think a single frame delay for cleanup would cause much trouble, or am I missing something here?

Reply:
I assume it's because we're waiting 1 frame to initiate the core instance.

https://github.com/locomotivemtl/locomotive-scroll/blob/10af1df02b77d2a2c94e22fee9fa92dcce3731b6/src/index.ts#L140-L167

Reply:
That's why I was trying to destroy it in the next frame.

The current flow is:

create new instance --> schedule core instance in next frame --> react immediately calls destroy (core instance isn't initialised yet so `this.coreInstance` is `undefined`) --> throws error

The flow I'm trying to achieve:

create new instance --> schedule core instance in next frame --> react immediately calls destroy --> schedule a core instance cleanup in next frame --> no error

Correct me if I'm wrong, but `this.coreInstance` should be available only in the next frame after the instance is initialised so I would guess cleaning it up outside rAF is **"assuming"** that the user doesn't destroy it immediately after initialisation:

```jsx
const locomotive = new Locomotive();
locomotive.destroy();
```

It kinda bothers me that we have to clean it up twice knowing that one of them won't have any effect. But it doesn't harm so I guess I'm fine with that.

Let me know what do you think so I'll apply the above suggestion if all looks fine.

Reply:
Typically there's no need to wait for 1 frame to destroy. It's only in the case of React or when there's an instant destroy that you should wait for 1 frame. In this case due to the `useEffect` running twice at init. I see no inconvenient calling the `this.coreInstance?.destroy()` two times

Reply:
Okay. Applying your suggestion then.